### ISSUE TITLE
DirectML (torch-directml): nanmedian fallback returns empty tensor on DirectML (privateuse) — recommend Windows ML migration / CPU median fallback

### ISSUE THREAD

**Issue by krishkal** [setiastro/setiastrosuitepro#71]
## Summary
On Windows with the DirectML backend (`torch-directml`) `torch.nanmedian` can fall back to CPU and return an empty tensor (shape `[0]`) for otherwise valid inputs. This breaks stacking/rejection algorithms that rely on robust reductions. DirectML is in maintenance mode; please consider adding a CPU-median fallback for DirectML users and/or evaluating Windows ML / ONNX Runtime on Win11 as a longer-term GPU path.

---

## Environment
- **OS:** Windows 11 25H2  
- **torch:** 2.4.1+cpu  
- **torch-directml:** 0.2.5.dev240914  
- **SASPro runtime python:** `C:\Users\(username)\AppData\Local\SASpro\runtime\py312\venv`  
- **Example tile shape used in repro:** (F, H, W, C) = (9, 512, 512, 3)

---

## Symptom / log excerpt
The log excerpt shows the following warning:
`UserWarning: The operator 'aten::nanmedian.dim_values' is not currently supported on the DML backend and will fall back to run on the CPU. ...`

However, when the fallback happens, it does not cleanly evaluate on the CPU. Instead nanmedian returns torch.Size([0]). The same inputs when directly evaluated using the CPU, it returns a valid (H,W,C) array. This is with DirectML on the observed device string: "privateuserone:0"

---

## Minimal Repro

The minimal code below conclusively proves that the problem is in DirectML when the Device is "privateuseone"

```Python
import torch, numpy as np, torch_directml
dev = torch_directml.device()
t = torch.from_numpy(np.random.rand(9,4,4,3).astype(np.float32)).to(dev)
print('torch', torch.__version__, 'device', dev)
try:
    m = torch.nanmedian(t, dim=0).values
    print('nanmedian ->', m.shape)
except Exception as e:
    print('nanmedian error:', repr(e))

# Compare with CPU/NumPy median:
print('numpy median ->', np.nanmedian(t.detach().cpu().numpy(), axis=0).shape)
```
### Observed Result

- torch.nanmedian on privateuserone yields torch.Size([0]) (empty tensor)
- np.nanmedian(..., axis=0) returns a valid (H,W,C) result.

---
## Recommended actions to consider

1. Short term - add explicit check and do a direct fallback to CPU when the DirectML device string contains "privateuse".
2. Longer-term -Evaluate Windows ML / ONNX Runtime (WinML provider) as the recommended GPU path on Windows 11 (Microsoft suggests WinML for Win11 24H2+). DirectML is not actively under development in favor of Windows ML/ONNX.

---
## Offer of help

I can do the short term fix suggested above, and provide code ready to PR. This will make the stacking code "work" by correctly falling back to the CPU for the unavailable nanmedian function.
